agent/grpc/public/services/peerstream/replication.go

```diff
@@ -240,12 +241,49 @@ func (s *Server) handleUpdateService(
 	sn structs.ServiceName,
 	pbNodes *pbservice.IndexedCheckServiceNodes,
 ) error {
+	_, peering, err := s.GetStore().PeeringRead(nil, state.Query{Value: peerName})
+	if err != nil {
+		return fmt.Errorf("failed to read peering: %w", err)
+	}
+	// an invariant here should be that a peering exists for PeeringRead
+	if peering == nil {
+		return fmt.Errorf("did not find peering %q", peerName)
+	}
+
+	logger := s.Logger.With("peerName", peerName, "peerID", peering.ID)
```

Comment:
This should probably have a `partition` field as well if the parameter is not the empty string.